pkg/controller/machine/controller.go

```diff
@@ -128,14 +143,38 @@ func (c *MachineControllerImpl) Reconcile(machine *clusterv1.Machine) error {
 	}
 	if exist {
 		glog.Infof("Reconciling machine object %v triggers idempotent update.", name)
-		return c.update(m)
+		err = c.update(m)
+		if err != nil {
+			requeueErr, ok := err.(*RequeueAfterError)
+			if ok {
+				glog.Infof("Actuator returned requeue after error: %v", requeueErr)
+				return c.enqueueAfter(machine, requeueErr.RequeueAfter)
+			}
+		}
+		return err
 	}
 	// Machine resource created. Machine does not yet exist.
 	glog.Infof("Reconciling machine object %v triggers idempotent create.", m.ObjectMeta.Name)
 	if err := c.create(m); err != nil {
-		glog.Warningf("unable to create machine %v: %v", name, err)
+		glog.Warningf("Unable to create machine %v: %v", name, err)
+		requeueErr, ok := err.(*RequeueAfterError)
+		if ok {
+			glog.Infof("Actuator returned requeue-after error: %v", requeueErr)
+			return c.enqueueAfter(machine, requeueErr.RequeueAfter)
+		} else {
+			return err
+		}
+	}
+	return nil
+}
+
+func (c *MachineControllerImpl) enqueueAfter(machine *clusterv1.Machine, after time.Duration) error {
+	key, err := cache.MetaNamespaceKeyFunc(machine)
+	if err != nil {
+		glog.Errorf("Couldn't get key for object %+v: %v", machine, after)
 		return err
 	}
+	c.queue.Queue.AddAfter(key, after)
```

Comment:
Does the queue have infinite size or can this call return an error?

Reply:
Looks like it must be infinite, no return values from the function.